otter/convergence/gathering.py

```diff
@@ -163,19 +165,45 @@ def _private_ipv4_addresses(server):
 
 def _servicenet_address(server):
     """
-    Finds the ServiceNet address for the given server.
+    Find the ServiceNet address for the given server.
     """
     return next((ip for ip in _private_ipv4_addresses(server)
                  if ip.startswith("10.")), "")
 
 
+def _lb_configs_from_metadata(server):
+    """
+    Construct a mapping of load balancer ID to :class:`ILBDescription` based
+    on the server metadata.
+    """
+    key_prefix = 'rax:autoscale:lb:'
+    desired_lbs = {}
+
+    # throw away any value that is malformed
+    for k in server.get('metadata', {}):
+        if k.startswith(key_prefix):
+            try:
+                config = json.loads(server['metadata'][k])
+
+                if config.get('type') != 'RackConnectV3':
+                    lbid = k[len(key_prefix):]
+                    desired_lbs[lbid] = CLBDescription(
+                        lb_id=lbid, port=config['port'])
+
+            except (KeyError, ValueError, TypeError):
```

Comment:
I'm wondering if this is a "put group in error state" kind of condition, since it almost certainly means the user messed with the metadata. However, that's not a call we have to make right now, so I'm creating a ticket for it.

https://github.com/rackerlabs/otter/issues/981